csharp/src/Apache.Arrow.Adbc/Tracing/FileExporter/FileExporter.cs

```diff
@@ -0,0 +1,272 @@
+﻿/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Diagnostics;
+using System.Text.Json;
+using System.Threading.Tasks;
+using System.Threading;
+using OpenTelemetry;
+using System.Collections.Concurrent;
+using System.IO;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Apache.Arrow.Adbc.Tracing.FileExporter
+{
+    internal class FileExporter : BaseExporter<Activity>
```

Comment:
FWIW, there's no product code which exposes this now and it's marked internal so it won't be accessible to any users.

Reply:
As we discussed separately, it might be better to expose this to end users as opposed to providing file tracing via driver options - which could inadvertently expose PII.